ping/ping_common.c

```diff
@@ -877,89 +884,14 @@ int gather_statistics(struct ping_rts *rts, uint8_t *icmph, int icmplen,
 	return 0;
 }
 
-static long llsqrt(long long a)
-{
-	long long prev = LLONG_MAX;
-	long long x = a;
-
-	if (x > 0) {
-		while (x < prev) {
-			prev = x;
-			x = (x + (a / x)) / 2;
-		}
-	}
-
-	return (long)x;
-}
-
 /*
  * finish --
  *	Print out statistics, and give up.
  */
 int finish(struct ping_rts *rts)
 {
-	struct timespec tv = rts->cur_time;
-	char *comma = "";
-
-	tssub(&tv, &rts->start_time);
-
-	putchar('\n');
-	fflush(stdout);
-	printf(_("--- %s ping statistics ---\n"), rts->hostname);
-	printf(_("%ld packets transmitted, "), rts->ntransmitted);
-	printf(_("%ld received"), rts->nreceived);
-	if (rts->nrepeats)
-		printf(_(", +%ld duplicates"), rts->nrepeats);
-	if (rts->nchecksum)
-		printf(_(", +%ld corrupted"), rts->nchecksum);
-	if (rts->nerrors)
-		printf(_(", +%ld errors"), rts->nerrors);
-
-	if (rts->ntransmitted) {
-#ifdef USE_IDN
-		setlocale(LC_ALL, "C");
-#endif
-		printf(_(", %g%% packet loss"),
-		       (float)((((long long)(rts->ntransmitted - rts->nreceived)) * 100.0) / rts->ntransmitted));
-		printf(_(", time %llums"), (unsigned long long)(1000 * tv.tv_sec + (tv.tv_nsec + 500000) / 1000000));
-	}
-
-	putchar('\n');
-
-	if (rts->nreceived && rts->timing) {
-		double tmdev;
-		long total = rts->nreceived + rts->nrepeats;
-		long tmavg = rts->tsum / total;
-		long long tmvar;
-
-		if (rts->tsum < INT_MAX)
-			/* This slightly clumsy computation order is important to avoid
-			 * integer rounding errors for small ping times. */
-			tmvar = (rts->tsum2 - ((rts->tsum * rts->tsum) / total)) / total;
-		else
-			tmvar = (rts->tsum2 / total) - (tmavg * tmavg);
+	ping_print_finish(rts);
 
-		tmdev = llsqrt(tmvar);
-
-		printf(_("rtt min/avg/max/mdev = %ld.%03ld/%lu.%03ld/%ld.%03ld/%ld.%03ld ms"),
-		       (long)rts->tmin / 1000, (long)rts->tmin % 1000,
-		       (unsigned long)(tmavg / 1000), (long)(tmavg % 1000),
-		       (long)rts->tmax / 1000, (long)rts->tmax % 1000,
-		       (long)tmdev / 1000, (long)tmdev % 1000);
-		comma = ", ";
```

Comment:
This is ugly, but actually very old change from 2010 964d0776f59e8ca5917c6368f90974446c2b2b32 (clearer would be to store `bool` and print `", "` below if set).

Reply:
I thought as much, but using a bool I will have to add if-conditionals and duplicate the `printf()` calls every time one can either occur with or without a comma.

Reply:
Never mind that, I got it now.